Samu/feature/260-feature-request-wallintroanimation

SProkopios opened the pull request:
closes #260

## 🔧 **Changes Made**

- created a component with an animation of the crumbling wall
- modified WallIntroAnimation  by introducing a child component, making modification easier
- added possibility to show it only on the first time (session storage), can be set via props if needed
- Jsdocs
- Storybooks
- Unit tests

---

## ✅ **Checklist Before Submission**

- **Functionality**: I have tested my code, and it works as expected. ✅
- **JSDoc**: I have added or updated JSDoc comments for all relevant code. ✅
- **Debugging**: No `console.log()` or other debugging statements are left. ✅
- **Clean Code**: Removed commented-out or unnecessary code. ✅
- **Tests**: Added new tests or updated existing ones for the changes made. ✅
- **Documentation**: Documentation has been updated (if applicable). 

---

## 📝 **Additional Information**

Provide any additional context or information that reviewers may need to know:

![doesitwork](https://github.com/user-attachments/assets/8555df58-609a-4671-89ab-41a250804654)

- **Known Issues**: [List any known issues or limitations]:
"P.S. try to avoid useEffects and setTimeouts if it possible" 
Sadly useEffects and setTimeouts still remains in the code.

Jonroi reviewed (changes requested):
I have to say that this is an overall good job; I like the way you made this component. If you have time to spare, you might notice there’s a way to simplify wall generation by combining the tile and image functions into one (not mandatory for this approval).

And would you embed this to the /admin page instead of the news page for now, so we can adjust it there. For the animation, I’d make it a little slower 0.020 looked much smoother to my eye.

Again, well done!

Jonroi reviewed (approved):
Code looks much simpler and more efficient now. Animation is much smoother now  (in my opinion) overall good job!

Skoivumaki reviewed (changes requested):
Recording (animations running 25% but bricks dont have time to fall on full speed):
![test2](https://github.com/user-attachments/assets/5ed661a5-ac6e-4fa4-a75a-553dc7764c48)
Animation itself looks cool, but there are some minor issues.

SProkopios commented on frontend-next-migration/src/shared/ui/v2/WallIntroAnimation/ui/WallIntroAnimation.tsx:
```diff
@@ -0,0 +1,66 @@
+            const timer = setTimeout(() => setIsLoaded(true), 50);
+            const hideTimer = setTimeout(() => setIsVisible(false), 1500);
+            if (renderOnce) {
+                sessionStorage.setItem('isRendered', 'true');
```
What would serve our purposes better? Local storage?

Skoivumaki replied:
> What would serve our purposes better? Local storage?

Correct 👍

SProkopios commented:
- deleted useless code
- set timout from 1500 to 2000
- changed sessionStorage to localStorage
- "Im curious why did you decide to wrap page content inside WallIntroAnimation, what do we gain from this?" i didint, its an old code. I fixed it so it no longer wraps children.

Jonroi commented:
I'm sorry, that was my poorly fixed merge conflict.

Skoivumaki reviewed (changes requested):
Bricks are still not wide enough. What I would do is to change the logic how bricks are set by devices width or just have them be as wide as 4k requires (if it doesnt make it behave weird)
![test](https://github.com/user-attachments/assets/c09abd80-a095-4ef8-ae47-34b582baf81e)
Responsively's 4k is kind of unrealistic, since UI is often automaticly scaled up on a real device like above:
![test2](https://github.com/user-attachments/assets/65c7610d-4c1c-4847-9ff5-dbdfe22e9908)

SProkopios commented:
Now bricks are set by devices width except for smaller screens, for smaller screens the same settings would have looked really ugly.

Skoivumaki reviewed (approved):
Nice work, now it looks great. I added it to main page since thats the most likely place for a user to land first time. Reviewers test on mainpage, there shouldnt be any issues but you never know😄

SProkopios commented:
Thats not how it should look like, Im going to inspect the code still!
